### PR TITLE
[TACHYON-1534] Fix flaky tachyon.shell.TfsShellTest#lsTest

### PR DESCRIPTION
**Pull request by dongche**
JIRA: https://tachyon.atlassian.net/browse/TACHYON-1534

Fix flaky tachyon.shell.TfsShellTest#lsTest. The clean and re-login is not a atomic operation, so that the re-login user sometimes is not expected. This patch put them into a synchronized block.

@apc999 , @sundapeng , would you like to take a look at this?

### CODE REVIEW

File: integration-tests/src/test/java/tachyon/shell/TfsShellTest.java [hunk 1]
```diff
@@ -506,14 +506,17 @@ public void locationNotExistTest() throws IOException {
...
   public void lsrTest() throws IOException, TachyonException {
-    // clear the loginUser
-    Whitebox.setInternalState(LoginUser.class, "sLoginUser", (String) null);
     MasterContext.getConf().set(Constants.SECURITY_GROUP_MAPPING,
         IdentityUserGroupsMapping.class.getName());
 
     FileInfo[] files = new FileInfo[4];
     String testUser = "test_user_lsr";
-    System.setProperty(Constants.SECURITY_LOGIN_USERNAME, testUser);
+    // clear the loginUser and re-login with new user
```
Comment (sundapeng) [inline integration-tests/src/test/java/tachyon/shell/TfsShellTest.java]: Thank @dongche for the fix, a minor comment is we can move `user login` to a private method.

**Comment by sundapeng** [comment]
@apc999 @dongche The pr is targeted to branch `master`, do we need to create a new pr for branch `authorization`

File: integration-tests/src/test/java/tachyon/shell/TfsShellTest.java (hunk 1 above)
Comment (dongche) [inline integration-tests/src/test/java/tachyon/shell/TfsShellTest.java]: Sounds good. Thanks

**Comment by dongche** [comment]
@sundapeng thanks for your review. Updated the patch to move them to a private method.
About the pr target, we will merge master to branch `authorization` later, then this fix would be in that branch.

**Comment by sundapeng** [comment]
LGTM, thank @dongche .
@apc999 Do you have any comment?

**Comment by aaudiber** [comment]
LGTM, thanks for the fix!

**Comment by apc999** [comment]
+1 LGTM

**Comment by yuzhiweilai** [comment]
Because of this,my test failed several times.Thanks for your fix.@sundapeng

**Pull request merged**